src/table/utils/find-visible-rows.ts

```diff
@@ -0,0 +1,42 @@
+import { TotalsPosition } from '../../types';
+
+const findStartIndex = (rows: HTMLCollectionOf<Element>, min: number, max: number): number => {
+  for (let i = 0; i < rows.length; i++) {
+    const rowRect = rows[i].getBoundingClientRect();
+    const center = rowRect.y + rowRect.height / 2;
+    if (center >= min && center <= max) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+const findEndIndex = (rows: HTMLCollectionOf<Element>, min: number, max: number): number => {
+  for (let i = rows.length - 1; i > -1; i--) {
+    const rowRect = rows[i].getBoundingClientRect();
+    const center = rowRect.y + rowRect.height / 2;
+    if (center >= min && center <= max) {
+      return i;
+    }
+  }
+  return -2;
```

Comment:
why -2?

Reply:
To make sure visibleRowEndIndex - visibleRowStartIndex + 1 is not bigger than 0. But I will refactor this to avoid confusion.